### DISCUSSION TITLE
how to run the performance tests?

### DISCUSSION THREAD

**Discussion by h3gui** [OpenDDS/OpenDDS#3715] [Q&A] [answered]
I do appreciate your help in providing the steps to run the latency tests so that I can capture the results in a similar table to https://opendds.org/perf/cent210803/latency_details.html
I was a bit lost in what resides under \OpenDDS-3.21\performance-tests especially those tests that require specifying options. For example -r rawData.log option when running the Priority test creates the file but never write anything to it despite the fact that the test passed.
Also, the performance_tests.lst file does not capture all the performance test cases and seems to point to tests with known deficiencies.
Thank you in advance,
Hafedh

**Comment by MayanNetwork** [comment]
hello h3gui,
I am trying to figure out how to run the latency tests, what are the steps?

**Comment by simpsont-oci** [comment, marked as answer]
The results generated in the link you provided are run by the "bench" performance test framework (see `performance-tests/bench`). The actual scenarios used there were generated to match the previous version of the bench performance tests (bench v1). Currently, this suite of scenarios are not available as part of the public repository. However the bench framework itself is designed to allow much more flexibility for designing and running tests, specifically because it allows for testing across multiple test "nodes" (real or virtual machines) running their own `node_controller` processes without requiring the workers to be precisely configured per machine. So creating your own meaningful latency scenarios should not be too difficult, assuming you're not trying to do anything too complicated.

Have a look at the bench (v2) documentation [here](https://opendds.readthedocs.io/en/latest/internal/bench2.html) and then look at the bench "echo" test that's run as part of our CI (in `performance-tests/bench`, you can run `/run_test.pl ci-echo`. This uses the scenario file `example/config/scenario/ci-echo.json` which in turn references the `ci-echo_client.json` and `ci-echo_server.json` worker configuration files in `example/config/worker`). This "echo" scenario is basically a simple echo server, and the "round trip latency" that's calculated from it is a safe latency calculation which avoids any issues from clock skew (assuming client and server are run on different machines).

Finally, If you are curious how the table you referenced is generated from a set of test bench test runs, you may also inspect the scripts used to generate the table and the individual plots. Those are located at `performance-tests/bench/legacy_plotting`. The README there gives a general description of how they are used after the test run has completed.
